examples/dom/WebReconciler.re

```diff
@@ -0,0 +1,218 @@
+          node |> Dom_html.element;
+        },
+        children,
+      }
+    );
+  };
+
+  let createElement = (~src, ~children, ()) => {
+    JsooReact.element(make(~src, JsooReact.listToElement(children)));
+  };
+};
+
+module Text = {
+  let component = JsooReact.nativeComponent("Text");
+
+  let make = (~text, children) => {
+    component((_: JsooReact.Hooks.empty) =>
+      {
+        make: () => {
+          let node = Dom_html.createSpan(document);
+          node##.innerHTML := Js.string(text);
+          node |> Dom_html.element;
+        },
+        configureInstance: (~isFirstRender as _, n) => {
+          /* TODO: Proper way to downcast? */
+          let node: Js.t(Dom_html.imageElement) = Obj.magic(n);
+          node##.innerHTML := Js.string(text);
+          node |> Dom_html.element;
+        },
+        children,
+      }
+    );
+  };
+
+  let createElement = (~text, ~children, ()) => {
+    JsooReact.element(make(~text, JsooReact.listToElement(children)));
+  };
+};
+
+module Button = {
+  let component = JsooReact.nativeComponent("Button");
+
+  let make = (~onPress, ~title, children) => {
+    component((_: JsooReact.Hooks.empty) =>
+      {
+        make: () => {
+          let node =
+            Dom_html.createButton(~_type=Js.string("button"), document);
+          let t = Js.string(title);
+          node##.title := t;
+          node##.innerHTML := t;
+          node##.onclick :=
+            Dom_html.handler(_e => {
+              onPress();
+              Js.bool(false);
+            });
+          node |> Dom_html.element;
+        },
+        configureInstance: (~isFirstRender as _, n) => {
+          /* TODO: Proper way to downcast? */
+          let node: Js.t(Dom_html.imageElement) = Obj.magic(n);
+          let t = Js.string(title);
+          node##.title := t;
+          node##.innerHTML := t;
+          node##.onclick :=
+            Dom_html.handler(_e => {
+              onPress();
+              Js.bool(false);
+            });
+          node |> Dom_html.element;
+        },
+        children,
+      }
+    );
+  };
+
+  let createElement = (~onPress, ~title, ~children, ()) => {
+    JsooReact.element(
+      make(~onPress, ~title, JsooReact.listToElement(children)),
+    );
+  };
+};
+
+/*
+    Step 4: Use the reconciler + native elements to create something great!
+ */
+type action =
+  | Increment
+  | Decrement;
+
+let reducer = (action, state) =>
+  switch (action) {
+  | Increment => state + 1
+  | Decrement => state - 1
+  };
+
+module CounterButtons = {
+  let component = JsooReact.component("Calculator");
+
+  let make = () =>
+    component(slots => {
+      let (count, dispatch, _slots: JsooReact.Hooks.empty) =
+        JsooReact.Hooks.reducer(~initialState=0, reducer, slots);
+      <View>
+        <Button title="Decrement" onPress={() => dispatch(Decrement)} />
+        <Text text={"Counter: " ++ str(count)} />
+        <Button title="Increment" onPress={() => dispatch(Increment)} />
```

Comment:
@wokalski I'm having some issues with the updates after dispatching an action from these buttons not showing in the UI. The DOM events are being read, and the `onPress` callback is called.

Do I have to somehow have a run loop or something that constantly flushes updates?

Reply:
I will look at it.